server/athenian/api/internal/jira.py

```diff
@@ -162,6 +173,60 @@ async def get_jira_installation_or_none(
         return None
 
 
+class JIRAEntitiesMapper:
+    """Maps various JIRA issue entities from their labels to their identifiers.
+
+    Issue priorities and issue types are mapped by this object.  Project are instead mapped
+    through JIRAConfig.
+    """
+
+    def __init__(
+        self,
+        priority_names_map: dict[str, list[str]],
+        types_map: dict[str, list[str]],
+    ) -> None:
+        """Build a mapper using the given dictionaries as mapping source."""
+        self._priority_names_map = priority_names_map
+        self._types_map = types_map
+
+    @classmethod
+    async def build(cls, jira_acc_id: int, mdb: DatabaseLike) -> JIRAEntitiesMapper:
+        """Build the mapper by loading data from DB."""
+        priority_stmt = (
+            sa.select(Priority.name, Priority.id).where(Priority.acc_id == jira_acc_id).distinct()
+        )
+        priority_rows = await mdb.fetch_all(priority_stmt)
+        priority_names_map = defaultdict(list)
+        for priority_name, priority_id in priority_rows:
+            priority_names_map[priority_name].append(priority_id)
+
+        type_stmt = (
+            sa.select(IssueType.name, IssueType.id)
+            .where(IssueType.acc_id == jira_acc_id)
+            .distinct()
+        )
+        types_map = defaultdict(list)
+        for type_, type_id in await mdb.fetch_all(type_stmt):
```

Comment:
Performance: must execute these two `mdb.fetch_all()` in parallel using `athenian.api.utils.gather()`